main.cpp

```diff
@@ -102,10 +103,24 @@ bool llama_model_load(const std::string & fname, llama_model & model, gpt_vocab
     {
         uint32_t magic;
         fin.read((char *) &magic, sizeof(magic));
-        if (magic != 0x67676d6c) {
+        if (magic == 0x67676d6c) {
+            fprintf(stderr, "%s: invalid model file '%s' (too old, regenerate your model files!)\n",
+                    __func__, fname.c_str());
+            return false;
+        }
+        if (magic != 0x67676d66) {
             fprintf(stderr, "%s: invalid model file '%s' (bad magic)\n", __func__, fname.c_str());
             return false;
         }
+
+        uint32_t format_version;
+        fin.read((char *) &format_version, sizeof(format_version));
+
+        if (format_version != 1) {
+            fprintf(stderr, "%s: invalid model file '%s' (unsupported format version %" PRIu32 ")\n",
```

Comment:
```suggestion
            fprintf(stderr, "%s: invalid model file '%s' (unsupported format version %" PRIu32 " - only version 1 is supported)\n",
```
Best to tell people not just that they have a wrong thing, but what they'd need to have instead.

(Also you might want to pull the supported version number out to a constant.)

Reply:
Oh lol I see `@j-f1` made exactly the same comment below, ignore me.